### PR TITLE
fix(GuildScheduledEvent): Use `if...else` pattern and handle partials

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
With the `GuildScheduledEvent` partial, it was possible for these properties to be overwritten. The `if...else` pattern prevents this.

Remembering that these can be swept out of the cache or limited, it was odd in the first place that this did not have partial handling, so that has been added.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -4911,9 +4920,9 @@ export interface ClientEvents {
     oldGuildScheduledEvent: GuildScheduledEvent | null,
```
Comment (jaw0r3k) [inline packages/discord.js/typings/index.d.ts:4920]:
```suggestion
 oldGuildScheduledEvent: GuildScheduledEvent | PartialGuildScheduledEvent | null,
```

This also can be partial, when cached structure was partial

File: packages/discord.js/typings/index.d.ts [hunk 2]
```diff
@@ -4908,12 +4917,12 @@ export interface ClientEvents {
   stickerUpdate: [oldSticker: Sticker, newSticker: Sticker];
   guildScheduledEventCreate: [guildScheduledEvent: GuildScheduledEvent];
   guildScheduledEventUpdate: [
-    oldGuildScheduledEvent: GuildScheduledEvent | null,
+   oldGuildScheduledEvent: GuildScheduledEvent | PartialGuildScheduledEvent | null,
```
Comment (jaw0r3k) [inline packages/discord.js/typings/index.d.ts:4920]:
```suggestion
    oldGuildScheduledEvent: GuildScheduledEvent | PartialGuildScheduledEvent | null,
    ```
Umm a little bit unfortunate edit

File: packages/discord.js/src/structures/GuildScheduledEvent.js [hunk 3]
```diff
@@ -50,11 +50,16 @@ class GuildScheduledEvent extends Base {
...
 
-    /**
-     * The name of the guild scheduled event
-     * @type {string}
-     */
-    this.name = data.name;
+    if ('name' in data) {
+      /**
+       * The name of the guild scheduled event
+       * @type {string}
```
Comment (jaw0r3k) [inline packages/discord.js/src/structures/GuildScheduledEvent.js:56]:
```suggestion
       * @type {?string}
```

File: packages/discord.js/src/structures/GuildScheduledEvent.js [hunk 4]
```diff
@@ -66,37 +71,59 @@ class GuildScheduledEvent extends Base {
...
 
-    /**
-     * The privacy level of the guild scheduled event
-     * @type {GuildScheduledEventPrivacyLevel}
-     */
-    this.privacyLevel = data.privacy_level;
+    if ('privacy_level' in data) {
+      /**
+       * The privacy level of the guild scheduled event
+       * @type {GuildScheduledEventPrivacyLevel}
```
Comment (jaw0r3k) [inline packages/discord.js/src/structures/GuildScheduledEvent.js:98]:
```suggestion
       * @type {?GuildScheduledEventPrivacyLevel}
```

File: packages/discord.js/src/structures/GuildScheduledEvent.js [hunk 5]
```diff
@@ -66,37 +71,59 @@ class GuildScheduledEvent extends Base {
...
 
-    /**
-     * The status of the guild scheduled event
-     * @type {GuildScheduledEventStatus}
-     */
-    this.status = data.status;
+    if ('status' in data) {
+      /**
+       * The status of the guild scheduled event
+       * @type {GuildScheduledEventStatus}
```
Comment (jaw0r3k) [inline packages/discord.js/src/structures/GuildScheduledEvent.js:109]:
```suggestion
       * @type {?GuildScheduledEventStatus}
```

File: packages/discord.js/src/structures/GuildScheduledEvent.js [hunk 6]
```diff
@@ -66,37 +71,59 @@ class GuildScheduledEvent extends Base {
...
 
-    /**
-     * The type of hosting entity associated with the scheduled event
-     * @type {GuildScheduledEventEntityType}
-     */
-    this.entityType = data.entity_type;
+    if ('entity_type' in data) {
+      /**
+       * The type of hosting entity associated with the scheduled event
+       * @type {GuildScheduledEventEntityType}
```
Comment (jaw0r3k) [inline packages/discord.js/src/structures/GuildScheduledEvent.js:120]:
```suggestion
       * @type {?GuildScheduledEventEntityType}
```

**Review by jaw0r3k — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**